src/libpriv/rpmostree-core.c

```diff
@@ -615,16 +614,18 @@ rpmostree_context_setup (RpmOstreeContext    *self,
   if (!dnf_context_setup (self->hifctx, cancellable, error))
     return FALSE;
 
-  /* NB: missing repo --> let hif figure it out for itself */
+  /* NB: missing repo --> let hif figure it out for itself. In that case, it's
+   * also perfectly fine to not have *any* repos at all (eg. local RPM case). */
```

Comment:
I'd expected to see some code here like "check whether the txn has only local RPMs".  So this is relying on the spec we generate not having a `repos` member?

Am I correct in that this is going to change the error message when one has an empty `/etc/yum.repos.d` and `rpm-ostree install foo` to "foo not found" instead of "No enabled repos"?

The latter is more informative...but eh, I think we can fix that by dumping more context information on errors (and possibly in the default case too).